packages/bpk-component-phone-input/src/BpkPhoneInput.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cssModules } from 'bpk-react-utils';
+import BpkInput, { INPUT_TYPES } from 'bpk-component-input';
+import BpkSelect from 'bpk-component-select';
+
+import STYLES from './BpkPhoneInput.scss';
+
+const getClassName = cssModules(STYLES);
+
+type ChildProps = { [string]: any };
+
+export type Props = {
+  dialingCode: string,
+  dialingCodeProps: { id: string, name: string, className?: string },
+  dialingCodes: Array<{ code: string, description: string }>,
+  id: string,
+  name: string,
+  onChange: (string, string) => void,
+  value: string,
+  className: ?string,
+  containerProps: ChildProps,
+  disabled: ?boolean,
```

Comment:
In general we don't have nulalble boolean props instead we explicitly make them `true` or `false` with `defaultProps`. In this case it seems like these should be `boolean` and have a default of `false`. Note this does not apply to `valid` beacuse it needs to be tri-state boolean